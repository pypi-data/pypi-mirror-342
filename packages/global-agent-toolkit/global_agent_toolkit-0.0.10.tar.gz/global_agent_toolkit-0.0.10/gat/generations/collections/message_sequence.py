from __future__ import annotations

from gat.generations.models.developer_message import DeveloperMessage
from gat.generations.models.message import Message
from rsb.collections.readonly_collection import ReadonlyCollection


class MessageSequence(ReadonlyCollection[Message]):
    def append(self, messages: list[Message]) -> MessageSequence:
        return MessageSequence(elements=list(self.elements) + list(messages))

    def without_developer_prompt(self) -> MessageSequence:
        return MessageSequence(
            list(
                filter(
                    lambda message: not isinstance(message, DeveloperMessage),
                    self.elements,
                )
            )
        )
