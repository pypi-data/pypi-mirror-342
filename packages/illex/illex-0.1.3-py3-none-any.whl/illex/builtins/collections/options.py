from illex.decorators.function import function
from illex.decorators.multi_param import multi_param_function


@function("options")
@multi_param_function
def handle_options(*args):
    result = {}
    for arg in args:
        if "=" not in arg:
            raise ValueError(f"Formato inv√°lido: {arg}")
        key, value = arg.split("=", 1)
        result[key.strip()] = value.strip()
    return result