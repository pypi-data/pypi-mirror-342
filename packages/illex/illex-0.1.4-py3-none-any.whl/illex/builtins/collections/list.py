from illex.decorators.function import function
from illex.decorators.multi_param import multi_param_function


@function("list")
@multi_param_function
def handle_list(*args):
    return list(args)
