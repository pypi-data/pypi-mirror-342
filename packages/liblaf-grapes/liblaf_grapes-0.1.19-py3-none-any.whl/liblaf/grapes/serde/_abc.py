import os
import tempfile
from pathlib import Path
from typing import Any

from liblaf import grapes


class AbstractSerializer:
    def load(self, fpath: str | os.PathLike[str], **kwargs) -> Any:
        if type(self).loads is not AbstractSerializer.loads:
            fpath: Path = grapes.as_path(fpath)
            return self.loads(fpath.read_text(), **kwargs)
        raise NotImplementedError

    def loads(self, data: str, **kwargs) -> Any:
        if type(self).load is not AbstractSerializer.load:
            with tempfile.NamedTemporaryFile("w") as fp:
                fp.write(data)
                return self.load(Path(fp.name), **kwargs)
        raise NotImplementedError

    def save(self, fpath: str | os.PathLike[str], data: Any, **kwargs) -> None:
        if type(self).saves is not AbstractSerializer.saves:
            fpath: Path = grapes.as_path(fpath)
            fpath.write_text(self.saves(data, **kwargs))
        raise NotImplementedError

    def saves(self, data: Any, **kwargs) -> str:
        if type(self).save is not AbstractSerializer.save:
            with tempfile.NamedTemporaryFile("w") as fp:
                self.save(Path(fp.name), data, **kwargs)
                return fp.read()
        raise NotImplementedError
