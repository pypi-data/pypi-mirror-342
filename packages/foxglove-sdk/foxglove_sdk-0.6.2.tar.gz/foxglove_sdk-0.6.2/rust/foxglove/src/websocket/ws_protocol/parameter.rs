//! Parameter types.

use std::collections::BTreeMap;

use serde::{Deserialize, Serialize};
use serde_with::base64::Base64;
use serde_with::serde_as;

/// A parameter type.
#[derive(Debug, Clone, Copy, PartialEq, Eq, Serialize, Deserialize)]
#[serde(rename_all = "snake_case")]
pub enum ParameterType {
    /// A byte array, encoded as a base64-encoded string.
    ByteArray,
    /// A decimal or integer value that can be represented as a `float64`.
    Float64,
    /// An array of decimal or integer values that can be represented as `float64`s.
    Float64Array,
}

/// A parameter value.
#[serde_as]
#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
#[serde(untagged)]
pub enum ParameterValue {
    /// A decimal or integer value.
    Number(f64),
    /// A boolean value.
    Bool(bool),
    /// A byte array, encoded as a base64-encoded string.
    String(#[serde_as(as = "Base64")] Vec<u8>),
    /// An array of parameter values.
    Array(Vec<ParameterValue>),
    /// An associative map of parameter values.
    Dict(BTreeMap<String, ParameterValue>),
}

/// Informs the client about a parameter.
#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
pub struct Parameter {
    /// The name of the parameter.
    pub name: String,
    /// The parameter type.
    #[serde(skip_serializing_if = "Option::is_none")]
    pub r#type: Option<ParameterType>,
    /// The parameter value.
    #[serde(skip_serializing_if = "Option::is_none")]
    pub value: Option<ParameterValue>,
}

impl Parameter {
    /// Creates a new parameter with no value or type.
    pub fn empty(name: impl Into<String>) -> Self {
        Self {
            name: name.into(),
            r#type: None,
            value: None,
        }
    }

    /// Creates a new parameter with a float64 value.
    pub fn float64(name: impl Into<String>, value: f64) -> Self {
        Self {
            name: name.into(),
            r#type: Some(ParameterType::Float64),
            value: Some(ParameterValue::Number(value)),
        }
    }

    /// Creates a new parameter with a float64 array value.
    pub fn float64_array(name: impl Into<String>, values: Vec<f64>) -> Self {
        Self {
            name: name.into(),
            r#type: Some(ParameterType::Float64Array),
            value: Some(ParameterValue::Array(
                values.into_iter().map(ParameterValue::Number).collect(),
            )),
        }
    }

    /// Creates a new parameter with a string value.
    pub fn string(name: impl Into<String>, value: impl Into<String>) -> Self {
        Self::byte_array(name, value.into().into_bytes())
    }

    /// Creates a new parameter with a byte array value.
    pub fn byte_array(name: impl Into<String>, data: impl Into<Vec<u8>>) -> Self {
        Self {
            name: name.into(),
            r#type: Some(ParameterType::ByteArray),
            value: Some(ParameterValue::String(data.into())),
        }
    }

    /// Creates a new parameter with a boolean value.
    pub fn bool(name: impl Into<String>, value: bool) -> Self {
        Self {
            name: name.into(),
            r#type: None,
            value: Some(ParameterValue::Bool(value)),
        }
    }

    /// Creates a new parameter with a dictionary value.
    pub fn dict(name: impl Into<String>, value: BTreeMap<String, ParameterValue>) -> Self {
        Self {
            name: name.into(),
            r#type: None,
            value: Some(ParameterValue::Dict(value)),
        }
    }
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn test_empty() {
        insta::assert_json_snapshot!(Parameter::empty("test"));
    }

    #[test]
    fn test_float() {
        insta::assert_json_snapshot!(Parameter::float64("f64", 1.23));
    }

    #[test]
    fn test_float_array() {
        insta::assert_json_snapshot!(Parameter::float64_array("f64[]", vec![1.23, 4.56]));
    }

    #[test]
    fn test_string() {
        insta::assert_json_snapshot!(Parameter::string("string", "howdy"));
    }

    #[test]
    fn test_byte_array() {
        insta::assert_json_snapshot!(Parameter::byte_array("byte[]", [0x10, 0x20, 0x30]));
    }

    #[test]
    fn test_bool() {
        insta::assert_json_snapshot!(Parameter::bool("bool", true));
    }

    #[test]
    fn test_dict() {
        insta::assert_json_snapshot!(Parameter::dict(
            "outer",
            maplit::btreemap! {
                "bool".into() => ParameterValue::Bool(false),
                "number".into() => ParameterValue::Number(1.23),
                "nested".into() => ParameterValue::Dict(
                    maplit::btreemap! {
                        "inner".into() => ParameterValue::Number(1.0),
                    }
                )
            }
        ));
    }
}
