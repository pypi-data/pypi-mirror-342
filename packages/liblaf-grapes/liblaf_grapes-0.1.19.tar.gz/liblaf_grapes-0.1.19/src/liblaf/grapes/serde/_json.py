import json as _json
import os
from pathlib import Path
from typing import Any, override

from liblaf import grapes

from ._abc import AbstractSerializer


class JSONSerializer(AbstractSerializer):
    @override
    def load(self, fpath: str | os.PathLike[str], **kwargs) -> Any:
        fpath: Path = grapes.as_path(fpath)
        with fpath.open() as fp:
            return _json.load(fp, **kwargs)

    @override
    def loads(self, data: str, **kwargs) -> Any:
        return _json.loads(data, **kwargs)

    @override
    def save(self, fpath: str | os.PathLike[str], data: Any, **kwargs) -> None:
        fpath: Path = grapes.as_path(fpath)
        with fpath.open("w") as fp:
            _json.dump(data, fp, **kwargs)

    @override
    def saves(self, data: Any, **kwargs) -> str:
        return _json.dumps(data, **kwargs)


json = JSONSerializer()
load_json = json.load
loads_json = json.loads
save_json = json.save
saves_json = json.saves
