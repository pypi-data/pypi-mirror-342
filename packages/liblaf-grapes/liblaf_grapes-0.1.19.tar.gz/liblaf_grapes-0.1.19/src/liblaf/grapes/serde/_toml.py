import os
from pathlib import Path
from typing import Any, override

import tomlkit

from liblaf import grapes

from ._abc import AbstractSerializer


class TOMLSerializer(AbstractSerializer):
    @override
    def load(self, fpath: str | os.PathLike[str], **kwargs) -> tomlkit.TOMLDocument:
        fpath: Path = grapes.as_path(fpath)
        with fpath.open() as fp:
            return tomlkit.load(fp, **kwargs)

    @override
    def loads(self, data: str, **kwargs) -> tomlkit.TOMLDocument:
        return tomlkit.loads(data, **kwargs)

    @override
    def save(self, fpath: str | os.PathLike[str], data: Any, **kwargs) -> None:
        fpath: Path = grapes.as_path(fpath)
        with fpath.open("w") as fp:
            tomlkit.dump(data, fp, **kwargs)

    @override
    def saves(self, data: Any, **kwargs) -> str:
        return tomlkit.dumps(data, **kwargs)


toml = TOMLSerializer()
load_toml = toml.load
loads_toml = toml.loads
save_toml = toml.save
saves_toml = toml.saves
