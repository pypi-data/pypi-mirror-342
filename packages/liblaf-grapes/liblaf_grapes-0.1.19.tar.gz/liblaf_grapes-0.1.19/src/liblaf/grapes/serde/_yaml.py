import io
import os
from pathlib import Path
from typing import Any, override

from ruamel.yaml import YAML

from liblaf import grapes

from ._abc import AbstractSerializer


class YAMLSerializer(AbstractSerializer):
    @override
    def load(self, fpath: str | os.PathLike[str], **kwargs) -> Any:
        fpath: Path = grapes.as_path(fpath)
        yaml = YAML(**kwargs)
        with fpath.open() as fp:
            return yaml.load(fp)

    @override
    def loads(self, data: str, **kwargs) -> Any:
        stream = io.StringIO(data)
        yaml = YAML(**kwargs)
        return yaml.load(stream)

    @override
    def save(self, fpath: str | os.PathLike[str], data: Any, **kwargs) -> None:
        fpath: Path = grapes.as_path(fpath)
        yaml = YAML(**kwargs)
        with fpath.open("w") as fp:
            yaml.dump(data, fp)

    @override
    def saves(self, data: Any, **kwargs) -> str:
        stream = io.StringIO()
        yaml = YAML(**kwargs)
        yaml.dump(data, stream)
        return stream.getvalue()


yaml = YAMLSerializer()
load_yaml = yaml.load
loads_yaml = yaml.loads
save_yaml = yaml.save
saves_yaml = yaml.saves
