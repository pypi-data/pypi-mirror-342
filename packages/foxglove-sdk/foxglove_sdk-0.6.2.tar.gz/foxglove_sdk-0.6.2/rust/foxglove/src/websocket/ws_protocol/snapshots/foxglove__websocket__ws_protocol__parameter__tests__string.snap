---
source: rust/foxglove/src/websocket/ws_protocol/parameter.rs
expression: "Parameter::string(\"string\", \"howdy\")"
---
{
  "name": "string",
  "type": "byte_array",
  "value": "aG93ZHk="
}
